internal/provider/util.go

```diff
@@ -9,6 +9,15 @@ import (
 	"golang.org/x/crypto/ssh"
 )
 
+// Algorithm represents the type of private key algorithm currently supported
+type Algorithm string
```

Comment:
Opt: Could use `iota` enum pattern instead.

Reply:
I looked into switching to use this pattern, but the interfaced imposed by the SDK schema forces everything to end up a string anyway.

This is even more constrained by the fact that I'm adding the use of attribute validation in the schema:

```
ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(SupportedAlgorithmsString(), false)),
```

The alternative would be me writing validators dedicated to the `Algorithm` type, and it starts to feel like a lot of boilerplate. My goal was just to "nudge" future work away from having a bunch of copies of those strings around, and close the door on typos etc.

Reply:
PS Generics could helps with this kind of stuff a lot. Those validator functions we provide would be even more useful.